src/clj/rems/application/model.clj

```diff
@@ -673,6 +673,15 @@
                           (for [command (:disable-commands (get-config))]
                             {:permission command}))))
 
+(defn- enrich-attachments [application get-user]
+  (let [redacted-ids (set (for [event (:application/events application)
+                                attachment (:application/redacted-attachments event)]
+                            (:attachment/id attachment)))
+        redacted? #(contains? redacted-ids (:attachment/id %))]
+    (->> application
+         (transform [:application/attachments ALL :attachment/user] get-user)
+         (setval [:application/attachments ALL redacted? :attachment/redacted] true))))
```

Comment:
this was the biggest change, changes in short:
- added function to aggregate attachments from events in `rems.application.model/application-view` (many events can add attachments)
- redact event updates redacted application attachments with `:attachment/redacted`
- event attachments are merged in enrichment with `get-attachments-for-application` (form answers)
- attachments are additionally enriched with redact roles
- when redact command is processed, application privacy is not applied yet. here redact roles can be used together with user id to check if user has permission to redact, since application model contains all necessary roles and permissions. also redacted attachment must contain event id for now
- when application privacy is applied, redact roles are used together with user id to stamp `:attachment/can-redact` similar to command processing